### PR TITLE
Implement "embulkSystemProperty" to auto-generate "embulk.properties" in the Embulk home

### PR DESCRIPTION
**Pull request by dmikurube**
It adds a new command `embulkSystemProperty` in `installEmbulkRunSet`. This would be the base of upcoming features to process/generate `embulk.properties` from this Gradle plugin.

See an example below.

```
installEmbulkRunSet {
    embulkHome file("/home/user/.embulk")
    embulkSystemProperty "key", "value"
    artifact ...
    artifact ...
}
```

This `installEmbulkRunSet` would install Maven artifacts under `/home/user/.embulk/`, and generate `/home/user/.embulk/embulk.properties` with `key=value`.

### CODE REVIEW

File: src/main/java/org/embulk/gradle/runset/InstallEmbulkRunSet.java [hunk 1]
```diff
@@ -234,11 +248,47 @@ private static Map<Object, Object> castMap(final Map map) {
...
+            final File temporaryDir = this.getTemporaryDir();
+            final Path path;
+            try {
+                path = Files.createTempFile(temporaryDir.toPath(), "embulk.", ".properties");
+            } catch (final IOException ex) {
+                throw new UncheckedIOException(ex);
+            }
+
+            this.from(path.toFile(), copySpec -> {
+                copySpec.rename(path.getFileName().toString(), "embulk.properties");
```
Comment (hiroyuki-sato) [inline src/main/java/org/embulk/gradle/runset/InstallEmbulkRunSet.java:276]: Does this part work if a temporary directory and `$HOME` use the different file partitions?
(i.e. `/home` and `/var/tmp` use the different partitions)

man from `rename(2)`

```
       EXDEV  oldpath and newpath are not  on  the  same  mounted  filesystem.
              (Linux  permits  a  filesystem to be mounted at multiple points,
              but rename() does not work across different mount  points,  even
              if the same filesystem is mounted on both.)
```

**Review by hiroyuki-sato — COMMENTED** [review]
Basically, LGTM :+1: . Could you check my comment?

File: src/main/java/org/embulk/gradle/runset/InstallEmbulkRunSet.java (hunk 1 above)
Comment (dmikurube) [reply src/main/java/org/embulk/gradle/runset/InstallEmbulkRunSet.java:276]: This `rename` does NOT mean the 
`rename` system call. It's just Gradle's implementation in `Copy`.

Comment (hiroyuki-sato) [reply src/main/java/org/embulk/gradle/runset/InstallEmbulkRunSet.java:276]: Thanks!

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**